metrics/src/vespa/metrics/updatehook.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "updatehook.h"
+
+namespace metrics {
+
+MetricLockGuard::MetricLockGuard(std::mutex & mutex)
+    : _guard(mutex)
+{}
+
+bool
+MetricLockGuard::owns(const std::mutex & mutex) const {
+    return (_guard.mutex() == &mutex) && !_guard.owns_lock();
```

Comment:
Should this be `&& _guard.owns_lock();` instead?

Reply:
Agreed, this looks like an error in the PR.

Reply:
Yes, 2 bugs cancelling each other causing the tests to still pass.